example/app/axis-configuration.tsx

```diff
@@ -0,0 +1,346 @@
+import * as React from "react";
+import { useMemo } from "react";
+import { SafeAreaView, ScrollView, StyleSheet, View } from "react-native";
+import {
+  CartesianChart,
+  Line,
+  Scatter,
+  type XAxisSide,
+  type YAxisSide,
+} from "victory-native";
+import type { AxisLabelPosition } from "lib/src/types";
+import { useDarkMode } from "react-native-dark";
+import { InputSlider } from "example/components/InputSlider";
+import { InputSegment } from "example/components/InputSegment";
+import {
+  optionsInitialState,
+  optionsReducer,
+} from "example/hooks/useOptionsReducer";
+import { InputColor } from "example/components/InputColor";
+import { InputText } from "example/components/InputText";
+import inter from "../assets/inter-medium.ttf";
+import { appColors } from "./consts/colors";
+import { InfoCard } from "../components/InfoCard";
+import { descriptionForRoute } from "./consts/routes";
+
+const parseTickValues = (tickString?: string) =>
+  tickString
+    ?.split(",")
+    .map((v) => parseFloat(v))
+    .filter((v) => !isNaN(v));
+
+const DATA = (ticksX: number[], ticksY: number[]) => {
+  const maxY = Math.max(...ticksY);
+  const minY = Math.min(...ticksY);
+  const maxX = Math.max(...ticksX);
+  const minX = Math.min(...ticksX);
+  const dX = maxX - minX;
+  const dY = maxY - minY;
+
+  return Array.from({ length: 10 }, (_, index) => ({
+    day: minX + (dX * index) / 10,
+    sales: Math.random() * dY + minY,
+  }));
+};
+
+export default function AxisConfiguration(props: { segment: string }) {
+  const description = descriptionForRoute(props.segment);
+  const isDark = useDarkMode();
+  const [
+    {
+      fontSize,
+      chartPadding,
+      strokeWidth,
+      xAxisSide,
+      yAxisSide,
+      xLabelOffset,
+      yLabelOffset,
+      xTickCount,
+      yTickCount,
+      xAxisLabelPosition,
+      yAxisLabelPosition,
+      scatterRadius,
+      colors,
+      domainPadding,
+      curveType,
+      customXLabel,
+      customYLabel,
+      xAxisValues,
+      yAxisValues,
+    },
+    dispatch,
+  ] = React.useReducer(optionsReducer, {
+    ...optionsInitialState,
+    domainPadding: 10,
+    chartPadding: 0,
+    strokeWidth: 2,
+    xAxisValues: "0,2,4,6,8",
+    yAxisValues: "-1,0,1,2,4,6,8",
+    colors: {
+      stroke: isDark ? "#fafafa" : "#71717a",
+      xLine: isDark ? "#71717a" : "#ffffff",
+      yLine: isDark ? "#aabbcc" : "#ddfa55",
+      frameLine: isDark ? "#444" : "#aaa",
+      xLabel: isDark ? appColors.text.dark : appColors.text.light,
+      yLabel: isDark ? appColors.text.dark : appColors.text.light,
+      scatter: "#a78bfa",
+    },
+  });
+  const font = useFont(inter, fontSize);
+  const ticksX = useMemo(
+    () => parseTickValues(xAxisValues) ?? [0, 10],
+    [xAxisValues],
+  );
+  const ticksY = useMemo(
+    () => parseTickValues(yAxisValues) ?? [0, 10],
+    [yAxisValues],
+  );
+
+  const data = useMemo(() => DATA(ticksX, ticksY), [xAxisValues, yAxisValues]);
```

Comment:
updated. the warning is annoying to look at, but might actually be wrong since comparing references to each other for memo'd values can produce weird results, whereas here i'm doing string comparison. not a big deal tho, this is just generating random data within the confines of the given ticks